SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs

```diff
@@ -395,9 +399,20 @@ internal void StartInstance(params Type[] services)
             }
 
             // Enqueue a log indicating which services have been initialized
-            if (startServiceLog.Services.Count > 0)
+            if (serviceNames.Count > 0)
             {
-                _channel.EnqueueAsync(startServiceLog);
+                if (InstanceEnabled)
+                {
+                    Task.Run(async () => await _channel.EnqueueAsync(new StartServiceLog { Services = serviceNames }).ConfigureAwait(false));
```

Comment:
The state can change by the time you enter the run code and execute enqueueAsync.
In Rum we handle that case with a cancellation token.
Or you can simply use the lock in the task and check again for InstanceEnabled.
I am not sure why you use a task.run in the first place in this method as we don't do that for example for custom properties.

Reply:
`Task.Run` used to force run in the background and `_channel` will be disabled on state change as well, so that will be checked inside. But you are right, here it isn't necessary.